MaxText/maxengine.py

```diff
@@ -49,6 +49,7 @@ class DecodeState:
   generate_cache: jax.Array
   generate_cache_index: int
   generate_lengths: jax.Array
+  generated_token: jax.Array
```

Comment:
is it `generated_tokens`? BTW seems the fields of DecodeState doesn't match in the code?

Reply:
actually in the code, the "generated_tokens" are generated_lengths here, "tokens" are corresponding to generated_token here.

Reply:
Should the field name match to each other? Otherwise, it's not using the pre-defined fields?